qt/applications/workbench/workbench/utils/recentlyclosedscriptsmenu.py

```diff
@@ -0,0 +1,98 @@
+
+
+class RecentlyClosedScriptsMenu(QMenu):
+    def __init__(self, mainwindow):
+        super(RecentlyClosedScriptsMenu, self).__init__()
+        self.setTitle("Open Recently Closed Scripts")
+        self.aboutToShow.connect(self.repopulate_menu)
+        self.mainwindow = mainwindow
+
+    def repopulate_menu(self):
+        self.clear()
+        self.populate_menu()
+
+    def populate_menu(self):
+        # Check cache is present or don't do anything.
+        scripts = self.load_scripts_from_settings()
+
+        if len(scripts) > 0:
+            for script_path in scripts:
+                script_name = self.size_path_correctly(script_path)
+                new_action = create_action(parent=self.mainwindow,
+                                           text=script_name,
+                                           on_triggered=functools.partial(self.open_script, script_path))
+                self.addAction(new_action)
+        else:
+            self.addAction(create_action(parent=self.mainwindow,
+                                         text="No recently closed scripts found"))
+
+    @staticmethod
+    def size_path_correctly(path):
+        return path if len(path) < 33 else "..." + path[30:]
+
+    def open_script(self, path):
+        # Check if it exists, if it doesn't pop up small window saying sorry this doesn't exist, then remove it from
+        # stack. else pass it off to the script window to open.
+        if exists(path):
+            self.mainwindow.editor.open_file_in_new_tab(path)
+        else:
+            # Remove path from script settings, then warn user.
+            self.remove_script_from_settings(path)
+            QMessageBox().warning(None, "That script no longer exists!",
+                                  "Are all network drives properly mounted? or are there any network connectivity "
+                                  "problems?",
+                                  QMessageBox.Ok)
+
+    @staticmethod
+    def remove_script_from_settings(path):
+        if CONF.has(RECENT_SCRIPTS_KEY):
+            scripts = CONF.get(RECENT_SCRIPTS_KEY)
+            if path in scripts:
+                scripts.remove(path)
+                CONF.set(RECENT_SCRIPTS_KEY, scripts)
+
+    @staticmethod
+    def load_scripts_from_settings():
+        scripts = []
+        if CONF.has(RECENT_SCRIPTS_KEY):
+            scripts = CONF.get(RECENT_SCRIPTS_KEY)
+
+        # Ensure there are no duplicates, general sanitization.
+        scripts = list(set(scripts))
+
+        return scripts
+
+    @staticmethod
+    def add_script_to_settings(path):
+        if path is None or path == '':
+            return
+        scripts = []
+        if CONF.has(RECENT_SCRIPTS_KEY):
+            scripts = CONF.get(RECENT_SCRIPTS_KEY)
+
+        if path not in scripts:
+            scripts.insert(0, path)
+
+        if len(scripts) > RECENT_SCRIPT_MAX_NUMBER:
+            scripts.pop()
+
+        CONF.set(RECENT_SCRIPTS_KEY, scripts)
```

Comment:
Can line 98 be moved into the `if` clause of line 92? It seems to me that it only needs to be called if the new path was not already in scripts and has been inserted during this function call. Could also maybe use an `else: return` on the same statement to avoid the `len(scripts)` check if nothing has been inserted

Reply:
Agreed